### ISSUE TITLE
voicenote org-link type doesn't jump to time offsets

### ISSUE THREAD

**Issue by mcshicks** [bgutter/voicenotes2org#3]
On mine I corrected it by changing this 

 eval: (org-link-set-parameters "voicenote" :follow (lambda (content) (cl-multiple-value-bind (file seconds) (split-string content ":") (emms-play-file file) (sit-for 0.5) (emms-seek-to seconds)))) 

to this
 eval: (org-link-set-parameters "voicenote" :follow (lambda (content) (cl-multiple-value-bind (file seconds) (split-string content ":") (emms-play-file file) (sit-for 0.5) (emms-seek-to (string-to-number seconds)))))

i.e. convert the string seconds into an integer before passing to emms-seek-to

**Comment by bgutter** [comment]
Thanks for reporting this!

Weird, I'm not sure how the code manages to work on my machine (just verified that it does, in fact, work without the fix here).  Which version of emacs and emms are you using? Both of my machines are at 28.0.50 and 5.3 respectively.

Actually..I do see the error on one of the machines...but not the other...weird..

**Comment by bgutter** [comment]
Resolved in 3518c927489da0950a7a89dc56d3ec70c18f6f21

**Comment by mcshicks** [comment]
So my emms version is old, 4.2.  Emacs version 25.3.2.

I looked at emms-seek-to in my version and here

https://github.com/emacsmirror/emms/blob/master/lisp/emms.el
and
https://github.com/emacsmirror/emms/blob/master/lisp/emms-player-mplayer.el

and they look the same.  Actually I think the error happens here in the
format

(defun emms-player-mplayer-seek-to (sec)
  "Depends on mplayer's -slave mode."
  (process-send-string
   emms-player-simple-process-name
   (format "seek %d 2\n" sec)))

Also it's not that it doesn't play on mine.  It plays, it just doesn't
jump to the correct time.  Also just to make things more confusing this
is after I switched everything to mp3 format.  I tried I think maybe one
case with the wav file, but I did not pay attention to the jump, just
that it was playing the file.

Here's a debug trace.

Debugger entered--Lisp error: (error "Format specifier doesn’t match argument type")
  apply(debug error (error "Format specifier doesn’t match argument type"))
  edebug(error (error "Format specifier doesn’t match argument type"))
  format("seek %d 2\n" "30")
  (process-send-string emms-player-simple-process-name (format "seek %d 2\n" sec))
  emms-player-mplayer-seek-to("30")
  funcall(emms-player-mplayer-seek-to "30")
  (if (not seek) (error "Player does not know how to seek-to") (funcall seek seconds) (run-hook-with-args (quote emms-player-time-set-functions) seconds))
  (let ((seek (emms-player-get emms-player-playing-p (quote seek-to)))) (if (not seek) (error "Player does not know how to seek-to") (funcall seek seconds) (run-hook-with-args (quote emms-player-time-set-functions) seconds)))
  (if (not emms-player-playing-p) (error "Can't seek-to player, nothing playing right now") (let ((seek (emms-player-get emms-player-playing-p (quote seek-to)))) (if (not seek) (error "Player does not know how to seek-to") (funcall seek seconds) (run-hook-with-args (quote emms-player-time-set-functions) seconds))))
  emms-player-seek-to("30")
  emms-seek-to("30")




Brandon Guttersohn writes:

> Thanks for reporting this!
>
> Weird, I'm not sure how the code manages to work on my machine (just verified that it does, in fact, work without the fix here).  Which version of emacs and emms are you using? Both of mine are at 28.0.50 and 5.3 respectively.
>
> Actually..I do see the error on one of the machines...but not the other...weird..

-- 
Stephen Hicks

**Comment by bgutter** [comment]
Yeah, on my machine where the error /did/ occur, it had the same trace as yourself. And it also played the WAV, but, did not jump to the offset.

My primary machine (where I've mostly been refiling notes), on the other hand, reliably played from the correct offset. However, I did notice that if I set debug-on-error, it would hit a different (but possibly related?) error a few seconds after successfully playing from the correct offset.

So, beats me. It seems like there's a difference in the behavior of `format`, or a race condition caused by the `sit-for`, or something. But, in any case, should be fixed now.
